src/aliases.js

```diff
@@ -0,0 +1,143 @@
+}
+
+let aliases;
+
+const Aliases = {
+    saveAliases : () => {
+        if(!fs.existsSync(ALIASES_FILE.directory)){
+            fs.mkdirSync(ALIASES_FILE.directory);
+        }
+        writeFileAtomicSync(ALIASES_FILE.fileName, JSON.stringify(aliases));
+    },
+    loadAliases : (create = false) => {
+        if(create){
+            const defaults = JSON.stringify(defaultAliases, null, 2);
+            if(!fs.existsSync(ALIASES_FILE.directory)){
+                fs.mkdirSync(ALIASES_FILE.directory, {recursive: true});
+            }
+            writeFileAtomicSync(ALIASES_FILE.fileName, defaults);
+            aliases = defaultAliases;
+        }
+        if(!create && !fs.existsSync(ALIASES_FILE.fileName)){
+            Aliases.loadAliases(true);
+        }
+        try {
+            aliases = JSON.parse(fs.readFileSync(ALIASES_FILE.fileName, { encoding: "utf8" }));
+        } catch (err) {
+            console.warn('An error occurred when trying to load %s. %s', ALIASES_FILE.fileName, err.message);
+            throw err;
+        }
+    },
+    addAlias : (cmd, alias) => {
+        if(!Aliases.isCommand(cmd) || Aliases.isDisabled(cmd)){
+            return false;
+        }
+        if(Object.values(aliases).filter(x => x.includes(alias.startsWith("!") ? alias : "!" + alias)).length > 0){
+            return false;
+        }
+        if(!alias.startsWith("!")){
+            aliases[cmd].push("!" + alias);
+        } else {
+            aliases[cmd].push(alias);
+        }
+        Aliases.saveAliases();
+        return true;
+    },
+    removeAlias : (cmd, alias) => {
+        if(!Aliases.isCommand(cmd) || Aliases.isDisabled(cmd)){
+            return false;
+        }
+        if(!Object.values(aliases).filter(x => x.includes(alias)).length > 0){
+            return false;
+        }
+        if(!aliases[cmd].includes(alias)){
+            return false;
+        }
+        const indexOfAlias = aliases[cmd].indexOf(alias);
+        aliases[cmd].splice(indexOfAlias, 1);
+        return true;
+    },
+    isDisabled : (cmd) => {
+        return aliases[cmd].includes("disabled");
+    },
+    disableCommand: (cmd) => {
+        if(!Aliases.isCommand(cmd) || Aliases.isDisabled(cmd)){
+            return false;
+        }
+        aliases[cmd].push("disabled");
+        Aliases.saveAliases();
+        return true;
+    },
+    enableCommand: (cmd) => {
+        if(!Aliases.isCommand(cmd) || Aliases.isDisabled(cmd)){
+            aliases[cmd].pop();
+            Aliases.saveAliases();
+            return true;
+        }
+        return false;
+    },
+    isAlias : (cmd, message) => {
+        if(Aliases.isDisabled(cmd)){
+            return false;
+        }
+        return aliases[cmd].includes(message.split(' ')[0]);
+    },
+    resetCommand : (cmd) => {
+        if(Aliases.isCommand(cmd)){
+            aliases[cmd] = defaultAliases[cmd];
+            return true;
+        }
+        return false;
+    },
```

Comment:
Resetting the command can add aliases that already exist.
E.g. the following sequence of commands creates duplicate aliases:
`!removealias remove remove`
`!removealias remove leave`
`!addalias brb leave`
`!addalias brb remove`
`!resetcmd remove`
`!addcmd brb brb2`

I would suggest to either:

1. remove all default aliases of `cmd` from the other commands
2. or only reset the aliases of `cmd` that are not already assigned to other commands

I think (1) could make it more predictable, but i am not sure.

Reply:
This would be more work to implement, but I'd probably suggest a different option: refuse to reset a command if any of its default aliases are assigned to another command (and output an error saying why). I think this gets the best result for the user, where they don't accidentally wipe out an alias without realizing it *or* not wipe out an alias they wanted to wipe out.